### PR TITLE
FEA-77-Obtain list of nearest cities from api

### PR DESCRIPTION
**Pull request by AgataLudwiczynska-CC**

### CODE REVIEW

File: src/pages/City/components/NearestCitiesWeatherWidget/NearestCitiesWeatherWidget.tsx [hunk 1]
```diff
@@ -21,7 +22,10 @@ const NearestCitiesWeatherWidget = ({ loading, cityName }: Props) => {
     }
 
     if (cityName) {
-      return ([] as Place[]).map((city) => <CityWeatherShortcut key={city.id} cityName={city.tags.name} />);
+      return data?.map((city) => {
+        console.log(city.tags.name);
```
Comment (JakubLukaszCC) [inline src/pages/City/components/NearestCitiesWeatherWidget/NearestCitiesWeatherWidget.tsx]: **typo** remove console.log please

Comment (AgataLudwiczynska-CC) [reply src/pages/City/components/NearestCitiesWeatherWidget/NearestCitiesWeatherWidget.tsx]: Done :)

File: src/api/services/Overpass.ts [hunk 2]
```diff
@@ -25,4 +26,30 @@ export default {
...
+      ['cities', cityName],
+      async () => {
+        const res = await overpassClient.post('/interpreter', query);
+
+        const normalizedNearestPlaces = res.data.elements
+          .map((city: Place) => {
+            return { ...city, tags: { name: convertSpecialCharacters(city.tags.name) } };
+          })
+          .filter((city: Place) => city.tags.name !== cityName)
+          .slice(0, 6);
```
Comment (klaudia-blazyczek-cc) [inline src/api/services/Overpass.ts:49]: **Question** Is this line necessary? If we filter the cities in the line above we should get 6 places anyways

File: src/components/CityWeatherShortcut/CityWeatherShortcut.tsx [hunk 3]
```diff
@@ -27,7 +26,7 @@ const CityWeatherShortcut = ({ cityName }: Props) => {
           <Value>{roundedTemperature}</Value>
           <TemperatureSign>&deg;C</TemperatureSign>
         </Temperature>
-        <City>{data.location.name}</City>
+        <City>{cityName}</City>
```
Comment (klaudia-blazyczek-cc) [inline src/components/CityWeatherShortcut/CityWeatherShortcut.tsx]: **Question** Why did you change it. Is there any reason to do so?

File: src/mocks/handlers.ts [hunk 4]
```diff
@@ -64,13 +64,6 @@ const handlers = [
   rest.get(ipLookupApiUrl, (req, res, ctx) => {
     return res(ctx.status(200), ctx.json(ipLookupData), ctx.delay(500));
   }),
-
```
Comment (klaudia-blazyczek-cc) [inline src/mocks/handlers.ts:67]: **Nitpick** Maybe try to add new things at the end of file, because just by looking at PR I thought you deleted 2 handlers

File: src/pages/City/components/NearestCitiesWeatherWidget/NearestCitiesWeatherWidget.tsx [hunk 5]
```diff
@@ -21,7 +22,7 @@ const NearestCitiesWeatherWidget = ({ loading, cityName }: Props) => {
     }
 
     if (cityName) {
```
Comment (klaudia-blazyczek-cc) [inline src/pages/City/components/NearestCitiesWeatherWidget/NearestCitiesWeatherWidget.tsx]: **Suggestion** Add data to if statement instead of cityName `if(data)`. CityName is passed as parameter so this statement is always true right now. If you change it like I suggested, you can then delete `?` in return statement.

**Review by klaudia-blazyczek-cc — DISMISSED** [review]

File: src/api/services/Overpass.ts (hunk 2 above)
Comment (AgataLudwiczynska-CC) [reply src/api/services/Overpass.ts:49]: Unfortunately, sometimes this city is on the list, and sometimes not, so for example in the case of Wrocław there are 7 cities on the list

File: src/mocks/handlers.ts (hunk 4 above)
Comment (AgataLudwiczynska-CC) [reply src/mocks/handlers.ts:67]: These handlers were identical to the other two above.

File: src/api/services/Overpass.ts (hunk 2 above)
Comment (klaudia-blazyczek-cc) [reply src/api/services/Overpass.ts:49]: Oh, okay, I get it

File: src/mocks/handlers.ts (hunk 4 above)
Comment (klaudia-blazyczek-cc) [reply src/mocks/handlers.ts:67]: Okay. Then maybe I should pay more attention, because at first it looked like you have deleted them

File: src/components/CityWeatherShortcut/CityWeatherShortcut.tsx (hunk 3 above)
Comment (AgataLudwiczynska-CC) [reply src/components/CityWeatherShortcut/CityWeatherShortcut.tsx]: I'll change it, this is our leftover from application testing

**Review by klaudia-blazyczek-cc — APPROVED** [review]

File: src/api/services/Overpass.ts (hunk 2 above)
Comment (JakubLukaszCC) [inline src/api/services/Overpass.ts]: **question** I'm wondering if this is solvable but when we get 7 cities and there is no our city it will remove 1 city and it could be the biggest city around our city.

Comment (AgataLudwiczynska-CC) [reply src/api/services/Overpass.ts]: Overpass API returns these 7 cities very randomly, so we also thought we could randomly delete one city. But if we decide that such sorting is needed, it won't be difficult to solve this problem.

**Review by JakubLukaszCC — APPROVED** [review]

File: src/pages/City/components/NearestCitiesWeatherWidget/NearestCitiesWeatherWidget.test.tsx [hunk 6]
```diff
@@ -1,22 +1,45 @@
...
 import { CityWeatherShortcutLoaderTestIds } from 'components/CityWeatherShortcut/components/CityWeatherShortcutLoader';
 import { NearestCitiesWeatherWidget } from './NearestCitiesWeatherWidget';
 import { NearestCitiesWeatherWidgetTestIds } from './NearestCitiesWeatherWidgetTestIds';
 
 describe('NearestCitiesWeatherWidget', () => {
   it('should render', () => {
-    render(<NearestCitiesWeatherWidget loading={false} cityName={undefined} />);
+    render(
+      <NearestCitiesWeatherWidget
+        cityName={weatherSuccessResponse.location.name}
```
Comment (bednarskimichal) [inline src/pages/City/components/NearestCitiesWeatherWidget/NearestCitiesWeatherWidget.test.tsx]: please use destructing where it's possible

**Review by kamil-borecki-cc — APPROVED** [review]
